src/components/tabController/index.tsx

```diff
@@ -54,17 +58,24 @@ function TabController({
   items,
   onChangeIndex = _.noop,
   carouselPageWidth,
+  useSafeArea,
   children
 }: PropsWithChildren<TabControllerProps>) {
-  const [screenWidth, setScreenWidth] = useState<number>(Constants.windowWidth);
+
+  const getScreenWidth = () => {
+    const {left, right} = Constants.getSafeAreaInsets();
+    return Constants.windowWidth - (useSafeArea && Constants.isIphoneX ? left + right : 0);
+  };
+
+  const [screenWidth, setScreenWidth] = useState<number>(getScreenWidth());
```

Comment:
It's a little weird the method `getScreenWidth` is above the useState. 
I suggest moving the function out of the component code, nothing fancy, it can be top/bottom of the file

Reply:
Fixed